https://issues.redhat.com/browse/ACM-24933 part 2 (#8413)
open-cluster-management/rhacm-docs: secure_clusters/create_cluster_permission.adoc

[#create-cluster-permission]
= Creating a cluster permission 

> When the service account is created, it does not have any permission associated to it. To grant permissions to the new service account, use the `ClusterPermission` resource. Create the `ClusterPermission` resource in the managed cluster namespace on the hub cluster. Use the `ClusterPermission` resource to create roles, cluster roles resources on the managed clusters, and bind them to a service account through a `RoleBinding` or `ClusterRoleBinding` resource.

Reviewer: We should not be using "it" so much. I recommend:

By default, a service account with created without an associated permission.

> When the service account is created, it does not have any permission associated to it. To grant permissions to the new service account, use the `ClusterPermission` resource. Create the `ClusterPermission` resource in the managed cluster namespace on the hub cluster. Use the `ClusterPermission` resource to create roles, cluster roles resources on the managed clusters, and bind them to a service account through a `RoleBinding` or `ClusterRoleBinding` resource.

Author: ```suggestion
When the service account is created there are no associated permissions. To grant permissions to the new service account, use the `ClusterPermission` resource. Create the `ClusterPermission` resource in the managed cluster namespace on the hub cluster. Use the `ClusterPermission` resource to create roles, cluster roles resources on the managed clusters, and bind them to a service account through a `RoleBinding` or `ClusterRoleBinding` resource.
```

*Required access:* Cluster administrator

.Prerequisites

- If you want to bind to the `ManagedServiceAccount` resource, you must have the `ManagedServiceAccount` add-on. For more information about enabling the `ManagedServiceAccount` add-on, see link:../add-ons/configure_klusterlet_addons.adoc[Configuring klusterlet add-ons]. 

.Procedure 

Complete the following steps to create a `ClusterPermission` resource that grants permissions to a managed service account:

. To grant the `<managed-sa-sample>` service account permission to the sample mortgage application that is deployed to the `mortgage` namespace on `<managed-cluster>`, create a YAML file with the following content: 

> 

Reviewer: We need to introduce this YAML better. I would rather not have most of the file a YAML file with no explanation. 

We need to introduce the kind, then we should have separate steps about the roles, role bindings, etc... It's not good practice to give a large about of YAML in one step and say Create this YAML.

Reviewer: See here that the bindings were created in a step in 2.14 and pieces of the YAML were defined in a step or the intro: https://github.com/stolostron/rhacm-docs/blob/2.14_stage/secure_clusters/fine_grain_rbac_cli.adoc

We got away from that for some reason here?

Author: This is old content. I just didn't focus on updating all of the content because we were up against time.

Author: @fxiang1

Reviewer 2: Right I agree with @dockerymick , I thought we were just moving this from the ALC Gitops section. This is not something new we've introduced recently.

Reviewer: We have gotten away from pushing large YAML samples without directions for them, but now I understand that this is previous content. I suggest we introduce the pieces that make the cluster permission, without reorganizing. Going forward the team agrees that this is not best practice and that important pieces of YAML need to be implemented as part of a larger process for understanding.

Reviewer 2: Yep understood, going through our docs I see we have many large YAML samples without explanation. Maybe we need to create a separate story/task for each of the component teams to just check and add explanation steps for these?

Author: @fxiang1 yup that's a great idea. I will improve the examples in this PR and then focus on other areas in the tech debt/refresh doc issue

Reviewer: Wow, these changes are more than I even expected here for this PR. I thought we could just add some intro info and be good but you took it even further to improve it. Great job.

Reviewer: @fxiang1 I think for the newer info that is being used a lot, yes, we can improve that YAML presentation. We have as a team mostly been doing that in new content. I don't think we will have time to fix everything, but certainly highly used doc we can improve.

[#create-cluster-permission]
= Creating a cluster permission 

When the service account is created, it does not have any permission associated to it. To grant permissions to the new service account, use the `ClusterPermission` resource. Create the `ClusterPermission` resource in the managed cluster namespace on the hub cluster. Use the `ClusterPermission` resource to create roles, cluster roles resources on the managed clusters, and bind them to a service account through a `RoleBinding` or `ClusterRoleBinding` resource.

*Required access:* Cluster administrator

.Prerequisites

> - are there any prereqs for this? 

Reviewer 2: Yes, we need to add the following:

- If the you wish to bind to the `ManagedServiceAccount` then the ManagedServiceAccount addon needs to be enabled. For help enabling the addon, see https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_management_for_kubernetes/2.14/html-single/add-ons/index#configure-klusterlet-addons .

Reviewer 2: Looks like this prereq applies to create_cluster_permission_exist.adoc and create_cluster_permission_ref.adoc as well. I wonder if we can have an overall prereq section for all the cluster permission docs?

> - are there any prereqs for this? 

Author: @fxiang1

Reviewer 2: Yes, it's the same prereq as above. That's why I'm wondering instead of repeating the same prereq three times, should we just have one prereq?
<img width="815" height="539" alt="image" src="https://github.com/user-attachments/assets/272eb340-8355-4afa-9adf-fea465478358" />

Author: I like your suggestions. It is required to list the the prereqs for each Task file (procedure focused), according to the [Red Hat Style Guide](https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs). However, we can discuss this at a future Doc guild

    - apiGroups: ["storage.k8s.io"]
      resources: ["*"]
      verbs: ["list"]
  - namespace: mortgage
    rules:
    - apiGroups: ["apps"]
      resources: ["deployments"]
      verbs: ["get", "list", "create", "update", "delete", "patch"]
    - apiGroups: [""]
      resources: ["configmaps", "secrets", "pods", "services", "namespace"]
      verbs: ["get", "update", "list", "create", "delete", "patch"]
  clusterRole:
    rules:
    - apiGroups: ["*"]
      resources: ["*"]
      verbs: ["get", "list"]
  roleBindings:
  - namespace: default
    roleRef:
      kind: Role
    subject:
      apiGroup: authentication.open-cluster-management.io
      kind: ManagedServiceAccount
      name: <managed-sa-sample>
  - namespace: mortgage
    roleRef:
      kind: Role
    subject:
      apiGroup: authentication.open-cluster-management.io
      kind: ManagedServiceAccount
      name: <managed-sa-sample>
  clusterRoleBinding:
    subject:
      apiGroup: authentication.open-cluster-management.io
      kind: ManagedServiceAccount
      name: <managed-sa-sample>
----

. Save the YAML file and name the file, `cluster-permission.yaml`.

> //should the name match the name in the previous example?

Reviewer 2: The user can use any name, it doesn't have to match.

> 

Author: @fxiang1 ^

Reviewer 2: See my comment above.
<img width="857" height="250" alt="image" src="https://github.com/user-attachments/assets/dfa683f8-e971-4071-8542-523eefea62a0" />

Author: Ok, I will remove the instruction about what to name the file.
